### PR TITLE
Iterate through list of client ids and call GetClient instead of iterating through Sockets vector

### PR DESCRIPTION
**Pull request by Sunderlandkyl**
@lassoan I believe this is what we discussed

re: https://github.com/IGSIO/OpenIGTLinkIO/issues/60

### CODE REVIEW

**Comment by lassoan** [comment]
It looks good. I haven't checked the full source code, but it should be clearly documented at least with comments what are the variables that may be accessed from multiple threads and what methods are safe to be called from any thread.

File: Logic/igtlioConnector.cxx [hunk 1]
```diff
@@ -408,11 +405,17 @@ int igtlioConnector::ReceiveController()
...
   {
-  igtlioLockGuard<vtkMutexLock> lock(this->Mutex);
-  for (std::vector<Client>::iterator clientIt = this->Sockets.begin(); clientIt != this->Sockets.end(); ++clientIt)
+    std::vector<int> clientIds = this->GetClientIds();
+    for (std::vector<int>::iterator clientIdIt = clientIds.begin(); clientIdIt != clientIds.end(); ++clientIdIt)
     {
+      Client client = this->GetClient(*clientIdIt);
+      if (client.ID == -1)
+      {
+        continue;
```
Comment (adamrankin) [inline Logic/igtlioConnector.cxx:414]: this change will break if the vector changes during iteration, no? if the vector is shortened to a size smaller than the current pointer of clientIdIt line 409 will throw an out-of bounds exception, right?

if you are going to do it this way, do it by
```c++
for(std::vector<int>::sizse_type i = 0; i < clientIds.size(); ++i)
{
...
}
```

Comment (lassoan) [reply Logic/igtlioConnector.cxx:414]: clientIds is a copy

Comment (adamrankin) [reply Logic/igtlioConnector.cxx:414]: Edit: nevermind, looked at the get ids function

**Review by adamrankin — APPROVED** [review]

**Pull request merged**